File: 05_prefix_sums/passing_cars.py
def solution(A):
  east_cnt = 0
  pair_cnt = 0

  for i in A:
    if i == 0:
      east_cnt += 1
    else:
      pair_cnt += east_cnt

  if pair_cnt > 10**9:
    return -1
  else:
    return pair_cnt

File: 05_prefix_sums/passing_cars.cpp
#include <vector>
#include <algorithm>

using namespace std;

const int MAX = 1000000000;

int solution(vector<int> &a) {
  int n = a.size();
  int ret = 0;
  int cnt = 0;
  for (int i = 0; i < n; i++) {
    if (a[i] == 1){
      ret += cnt;
      if (ret > MAX)
        return -1;
    }
    else
      cnt++;
  }
  return ret;
}
